runner-setup/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Dockerfile to setup a new runner for CI service.
+# This file has all dependencies for running the following service(s):
+#   - gen_optimized.py
+
+# Use a lightweight base image
+FROM ubuntu:24.04
+
+# Set environment variables
+# Should probably use docker compose for this
+ENV RUNNER_USER=runner
+ENV RUNNER_HOME=/home/${RUNNER_USER}
+# Should probably use docker compose and to set these values:
+ENV RUNNER_VERSION=2.319.1
+ENV GITHUB_TOKEN=#<Github Token Here>
+ENV REPO_URL=https://github.com/dtcxzyw/llvm-opt-benchmark
+
+# Install necessary packages
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    ninja-build \
+    cmake \
+    sudo \
+    ccache \
+    python3-pip \
+    libkrb5-3 \
+    zlib1g-dev \
+    liblttng-ust1t64 \
+    libssl-dev \
+    libicu-dev \
+    cargo \
+    gawk \
+    bison \
+    wget \
+    flex \
+    curl \
+    jq \
+    git \
```

Comment:
Do we have `perf` installed in the base image?

Reply:
Shoot, probably not. Will need to restart.

Reply:
Will probably also need to enable userland events collection.